### ISSUE TITLE
Throw exception when contig consists of only a single interval in gCNV tools.

### ISSUE THREAD

**Issue by samuelklee** [broadinstitute/gatk#5852]
See https://gatkforums.broadinstitute.org/gatk/discussion/23793/python-error-using-postprocessgermlinecnvcalls

**Comment by samuelklee** [comment]
@mwalker174 @asmirnov239 decided not to tackle this yet.  There are a few options: 1) warn and/or filter out such contigs in PreprocessIntervals (not a fan of this), 2) filter in FilterIntervals, 3) warn/fail/filter a little earlier at DetermineGermlineContigPloidy (and also GermlineCNVCaller to be safe), 4) warn/filter at PostprocessGermlineCNVCalls, 5) fix the theano code to treat such contigs specially (haven't looked closely at it, but probably has something to do with patching the foward-backward code to handle such cases.  Probably option 5 is the right answer, but only if the inferences for such single-interval contigs is at all meaningful.  Otherwise I'm inclined to do option 2 and add some warnings/exceptions downstream.

**Comment by fgvieira** [comment]
I am also getting this error on chromosome 11:
```
11:49:44.992 INFO gcnvkernel.postprocess.viterbi_segmentation - Segmenting contig (5/12) (contig name: 9)...
11:49:44.996 INFO gcnvkernel.postprocess.viterbi_segmentation - Segmenting contig (6/12) (contig name: 11)...

Stderr: Traceback (most recent call last):
  File "/ngc/projects/gm/data/resources/envs/conda/ngs_gatk_cnv/4.1.6.0/lib/python3.6/site-packages/theano/compile/function_module.py", line 903, in __call__
    self.fn() if output_subset is None else\
  File "/ngc/projects/gm/data/resources/envs/conda/ngs_gatk_cnv/4.1.6.0/lib/python3.6/site-packages/theano/scan_module/scan_op.py", line 963, in rval
    r = p(n, [x[0] for x in i], o)
  File "/ngc/projects/gm/data/resources/envs/conda/ngs_gatk_cnv/4.1.6.0/lib/python3.6/site-packages/theano/scan_module/scan_op.py", line 952, in p
    self, node)
  File "scan_perform.pyx", line 215, in theano.scan_module.scan_perform.perform
NotImplementedError: We didn't implemented yet the case where scan do 0 iteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/segment_gcnv_calls.6491270870870970325.py", line 79, in <module>
    viterbi_engine.write_copy_number_segments()
  File "/ngc/projects/gm/data/resources/envs/conda/ngs_gatk_cnv/4.1.6.0/lib/python3.6/site-packages/gcnvkernel/postprocess/viterbi_segmentation.py", line 234, in write_copy_number_segments
    for segment in self._viterbi_segments_generator():
  File "/ngc/projects/gm/data/resources/envs/conda/ngs_gatk_cnv/4.1.6.0/lib/python3.6/site-packages/gcnvkernel/postprocess/viterbi_segmentation.py", line 160, in _viterbi_segments_generator
    log_prior_c, log_trans_contig_tcc, copy_number_log_emission_contig_tc)
  File "/ngc/projects/gm/data/resources/envs/conda/ngs_gatk_cnv/4.1.6.0/lib/python3.6/site-packages/gcnvkernel/models/theano_hmm.py", line 88, in perform_forward_backward
    prev_log_posterior_tc, admixing_rate, temperature)))
  File "/ngc/projects/gm/data/resources/envs/conda/ngs_gatk_cnv/4.1.6.0/lib/python3.6/site-packages/theano/compile/function_module.py", line 917, in __call__
    storage_map=getattr(self.fn, 'storage_map', None))
  File "/ngc/projects/gm/data/resources/envs/conda/ngs_gatk_cnv/4.1.6.0/lib/python3.6/site-packages/theano/gof/link.py", line 325, in raise_with_op
    reraise(exc_type, exc_value, exc_trace)
  File "/ngc/projects/gm/data/resources/envs/conda/ngs_gatk_cnv/4.1.6.0/lib/python3.6/site-packages/six.py", line 692, in reraise
    raise value.with_traceback(tb)
  File "/ngc/projects/gm/data/resources/envs/conda/ngs_gatk_cnv/4.1.6.0/lib/python3.6/site-packages/theano/compile/function_module.py", line 903, in __call__
    self.fn() if output_subset is None else\
  File "/ngc/projects/gm/data/resources/envs/conda/ngs_gatk_cnv/4.1.6.0/lib/python3.6/site-packages/theano/scan_module/scan_op.py", line 963, in rval
    r = p(n, [x[0] for x in i], o)
  File "/ngc/projects/gm/data/resources/envs/conda/ngs_gatk_cnv/4.1.6.0/lib/python3.6/site-packages/theano/scan_module/scan_op.py", line 952, in p
    self, node)
  File "scan_perform.pyx", line 215, in theano.scan_module.scan_perform.perform
NotImplementedError: We didn't implemented yet the case where scan do 0 iteration
Apply node that caused the error: forall_inplace,cpu,scan_fn}(Elemwise{minimum,no_inplace}.0, InplaceDimShuffle{0,2,1}.0, Subtensor{int64:int64:int64}.0, IncSubtensor{InplaceSet;:int64:}.0, Shape_i{0}.0)
Toposort index: 95
Inputs types: [TensorType(int64, scalar), TensorType(float64, 3D), TensorType(float64, matrix), TensorType(float64, matrix), TensorType(int64, scalar)]
Inputs shapes: [(), (0, 6, 6), (0, 6), (2, 6), ()]
Inputs strides: [(), (288, 8, 48), (48, 8), (48, 8), ()]
Inputs values: [array(0), array([], shape=(0, 6, 6), dtype=float64), array([], shape=(0, 6), dtype=float64), 'not shown', array(6)]
Inputs type_num: [7, 12, 12, 12, 7]
Outputs clients: [[Subtensor{int64:int64:int8}(forall_inplace,cpu,scan_fn}.0, ScalarFromTensor.0, ScalarFromTensor.0, Constant{1})]]

        at org.broadinstitute.hellbender.utils.python.PythonExecutorBase.getScriptException(PythonExecutorBase.java:75)
        at org.broadinstitute.hellbender.utils.runtime.ScriptExecutor.executeCuratedArgs(ScriptExecutor.java:130)
        at org.broadinstitute.hellbender.utils.python.PythonScriptExecutor.executeArgs(PythonScriptExecutor.java:170)
        at org.broadinstitute.hellbender.utils.python.PythonScriptExecutor.executeScript(PythonScriptExecutor.java:151)
        at org.broadinstitute.hellbender.utils.python.PythonScriptExecutor.executeScript(PythonScriptExecutor.java:121)
        at org.broadinstitute.hellbender.tools.copynumber.PostprocessGermlineCNVCalls.executeSegmentGermlineCNVCallsPythonScript(PostprocessGermlineCNVCalls.java:620)
        at org.broadinstitute.hellbender.tools.copynumber.PostprocessGermlineCNVCalls.generateSegmentsVCFFileFromAllShards(PostprocessGermlineCNVCalls.java:395)
        at org.broadinstitute.hellbender.tools.copynumber.PostprocessGermlineCNVCalls.onTraversalSuccess(PostprocessGermlineCNVCalls.java:366)
        at org.broadinstitute.hellbender.engine.GATKTool.doWork(GATKTool.java:1050)
        at org.broadinstitute.hellbender.cmdline.CommandLineProgram.runTool(CommandLineProgram.java:139)
        at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMainPostParseArgs(CommandLineProgram.java:191)
        at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:210)
        at org.broadinstitute.hellbender.Main.runCommandLineProgram(Main.java:163)
        at org.broadinstitute.hellbender.Main.mainEntry(Main.java:206)
        at org.broadinstitute.hellbender.Main.main(Main.java:292)
Using GATK jar /ngc/tools/dtutools/gatk/4.1.6.0/gatk-package-4.1.6.0-local.jar
Running:
    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar gatk-package-4.1.6.0-local.jar PostprocessGermlineCNVCalls --model-shard-path shard_path-model --contig-ploidy-calls contig_ploidy-calls --calls-shard-path shard_path-calls --sample-index 0 --autosomal-ref-copy-number 2 --allosomal-contig X --allosomal-contig Y --output-denoised-copy-ratios out.gatk.denoised_copy_ratios --output-genotyped-intervals out.gatk.genotyped_intervals --output-genotyped-segments out.gatk.vcf.gz
```

Comparing with the other chromosomes, the only difference seems to be that the BED file only has a single interval in this chromosome:
```
[...]
9       98002930        98003025        NP_000127.2     .       -
9       98009713        98009798        NP_000127.2     .       -
9       98011408        98011573        NP_000127.2     .       -
11      22646231        22647356        NP_073562.1     .       -
13      32890597        32890664        NP_000050.2     .       +
13      32893213        32893462        NP_000050.2     .       +
13      32899212        32899321        NP_000050.2     .       +
[...]
```

and not sure option 5) is possible:
https://github.com/Theano/Theano/issues/4902

**Comment by samuelklee** [comment]
Thanks for the report, @fgvieira. I’d probably just drop that interval manually if I were you.

Just to clarify for whoever ends up tackling this, by option 5 I meant changing the relevant theano code/calls in our gcnvkernel package to handle this case appropriately (as opposed to code in the theano package itself).  But again, if inferences are not reasonable, then we may want to filter upstream.

**Comment by asmirnov239** [comment]
Closed by #6559
